### PR TITLE
DataLinks: Default to percentEncoding

### PR DESCRIPTION
**Pull request by ryantxu**
When creating datalinks with template variables, we should percentencode the values unless otherwise specified.

TODO:
- [x] tests

### CODE REVIEW

**Comment by ryantxu** [comment]
@torkelo / @dprokop -- I was surprised that the formatting registry is in scenes, that feels like something that should be in `@grafana/data` 🤔

**Comment by torkelo** [comment]
@ryantxu think you missed the most important place (not sure if link_srv or linkSupplier in app is used anywhere), 
I think this is the place datalinks get interpolated 
https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/field/fieldOverrides.ts#L468

**Comment by torkelo** [comment]
@ryantxu yea, could move it there but so far it's only in core (before scenes), and was never exposed

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -464,8 +463,8 @@ export const getLinksSupplier =
         : link.url;
 
       if (href) {
-        locationUtil.assureBaseUrl(href.replace(/\n/g, ''));
-        href = replaceVariables(href, variables);
+        href = locationUtil.assureBaseUrl(href.replace(/\n/g, ''));
```
Comment (ryantxu) [inline packages/grafana-data/src/field/fieldOverrides.ts:466]: ^^^ unrelated, but seems like this was a noop from your last change @dprokop

**Comment by ryantxu** [comment]
Moving schema from https://github.com/grafana/grafana/pull/64846

**Review by mckn — APPROVED** [review]
LGTM

**Pull request merged**